barman/retention_policies.py

```diff
@@ -182,9 +241,20 @@ def _backup_report(self, source):
                     self._first_backup = bid
                 else:
                     report[bid] = BackupInfo.OBSOLETE
+                    # If backup is postgres with incremental backup children
+                    # create tree o backups and make them all OBSOLETE.
+                    if self.has_children(backups[bid]):
+                        self.propagate_annotation_to_children(
+                            backup_info=backups[bid],
+                            report=report,
+                            annotation=BackupInfo.OBSOLETE,
+                        )
+                    else:
+                        _logger.debug("No children to propagate annotation...")
                 i = i + 1
             else:
                 report[bid] = BackupInfo.NONE
+
```

Comment:
Maybe we could change things a bit and process like this:

* Skip the backup in the beginning, as you are already doing -- maybe by using the new method of `LocalBackupInfo` that I mentioned in the previous comment;
* During the loop, try to identify the expected status for the children;
* After the loop, loop over `walk_backups_tree` and set the status accordingly